### DISCUSSION TITLE
Sometimes messages doesn't show and sub emotes are missing

### DISCUSSION THREAD

**Discussion by luizpheelipe** [Chatterino/chatterino2#4864] [Q&A] [answered]
Title is self explanatory, I have been this problem for a while and couldn't find a similar post solution.
I noticed that when having the twitch chat on the browser and chatterino side by side, the message problme is just sometimes, but the emotes are missing, like 2-3 from some streamers I am subbed to, emotes are shown in the browser but not in chatterino client.

**Comment by Felanbird** [comment]
The reason messages don't send is due to the fact that when you send a message from browser chat, Twitch will show as if it sent to chat, even if it didn't. On Chatterino we only show you if your message was bounced back from Twitch (meaning they received and accepted it).

If a sent message is missing a Twitch emote it's due to the fact that Twitch did not send us the correct emote tags to recognize that you actually sent an emote, and not text. 

![image](https://github.com/Chatterino/chatterino2/assets/41973452/e3bcf36e-dc01-4f68-9954-63fb91717b10)

For the first message, a Twitch emote is sent and Twitch send back the following  for that emote `emotes=emotesv2_2ca05f01d22a4756a11996adeee49138:0-7;` 

The second message was not a real emote, so Twitch sent back `@emotes=;`

As a note tho, we can't live-update your subscriptions, so if your sub was to run out, the emote & `:` menu might still show the emotes, but Twitch correctly does not send the emote tags, as you are no longer subbed.

**Reply by luizpheelipe** [reply]
It is not messages being sent by me are message others are typing and sometimes tagging with @ or just autocomplete with "tab", they don't show for me at all. It is not only messages of me being tagged, just randomly won't show.

And the emotes problem is that they don't show on the client tab of subscriptions, it is not for the lack of updates, some of the emotes are there since forever and I just realized they existed because I checked the browser instead of using the chatterino client.
img exemple: **Left** is chatterino client **Right** is twitch's
![vRpryhF](https://github.com/Chatterino/chatterino2/assets/93266049/8d6014e8-0ec4-4fbf-ac70-fd85303d4db9)

It is not just this particular channel it is with others as well.

**Reply by Felanbird** [reply]
Oh, those are follower emotes which only show up in the channel they are from on Chatterino due to the way they were originally implemented, it's mentioned in the `Commonly Reported Issues`
![image](https://github.com/Chatterino/chatterino2/assets/41973452/c306ad8e-04b5-421a-9af6-bade40a4b03b)
![image](https://github.com/Chatterino/chatterino2/assets/41973452/30b63d7b-926c-476f-b3a3-374a5283ecc2)

**Comment by 8thony** [comment, marked as answer]
- Fixed since #4922
